okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java

```diff
@@ -260,9 +260,14 @@ private void connectSocket(int connectTimeout, int readTimeout, Call call,
 
   private void establishProtocol(ConnectionSpecSelector connectionSpecSelector,
       int pingIntervalMillis, Call call, EventListener eventListener) throws IOException {
-    if (route.address().sslSocketFactory() == null) {
-      protocol = Protocol.HTTP_1_1;
+    if (route.address().protocols().contains(Protocol.H2C)) {
```

Comment:
It seems like this check could be a subset of the "if (route.address().sslSocketFactory() == null) " case

meaning 0 cost in the more common https case, and clearer that this is a special case of clear text.  Thoughts?